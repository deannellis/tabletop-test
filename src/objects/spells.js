export default [
    {
        id: '0000',
        level: 1,
        name: 'Charm Person',
        range: '30ft',
        type: 'magic-user',
        duration: 'special',
        shortDescription: 'This spell makes a humanoid creature regard the caster as its trusted friend and ally.',
        longDescription: 'This spell makes a humanoid creature of 4 hit dice or less regard the caster as its trusted friend and ally. And more...'
    },
    {
        id: '0001',
        level: 1,
        name: 'Detect Magic',
        range: '60ft',
        type: undefined,
        duration: '2 turns',
        shortDescription: 'The caster of this spell is able to detect magic objects or creatures.',
        longDescription: 'The caster of this spell is able to detect enchanted or enspelled objects or creatures within the given range by sight, seeing them surrounded by a pale glowing light. And more...'
    },
    {
        id: '0002',
        level: 1,
        name: 'Floating Disc',
        range: '10ft',
        type: 'magic-user',
        duration: '5 turns + 1/level',
        shortDescription: 'This spell creates an invisible plane of force for carrying loads.',
        longDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads. And more...'
    },
    {
        id: '0003',
        level: 1,
        name: 'Hold Portal',
        range: '100ft + 10ft/level',
        type: 'magic-user',
        duration: '1 round/level',
        shortDescription: 'This spell magically holds shut a portal (door, gate, window, etc)',
        longDescription: 'This spell magically holds shut a door, gate, window, or shutter of wood, metal, or stone. And more...'
    },
    {
        id: '0004',
        level: 1,
        name: 'Light',
        range: '120ft',
        type: undefined,
        duration: '6 turns + 1/level',
        shortDescription: 'This spell creates a light equal to torchlight',
        longDescription: 'This spell creates a light equal to torchlight which illuminates a 30ft radius area (and provides dim light for an additional 20) around the target location or object. And more...'
    },
    {
        id: '0005',
        level: 1,
        name: 'Cure Light Wounds',
        range: 'touch',
        type: 'cleric',
        duration: 'instantaneous',
        shortDescription: 'The caster heals 1d6+1 hit points of damage',
        longDescription: 'With this spell the caster heals 1d6+1 hit points of damage by laying his or her hand upon the injured creature.'
    }
]

// const sampleSpell = {
//     id: 0000,
//     level: 1,
//     name: 'Floating Disc',
//     range: '0',
//     type: 'magic-user',
//     duration: '5 turns + 1/level',
//     shortDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads.',
//     longDescription: 'This spell creates an invisible, slightly concave circular plane of force for carrying loads.'
// }