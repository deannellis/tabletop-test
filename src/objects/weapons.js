// export default {
//     0000: {
//         damage: 6,
//         name: 'Hand Axe',
//         price: 400, //copper pieces
//         range: 1,
//         size: 1, // 1 = Small, 2 = Medium, 3 = Large
//         weight: 5 //pounds
//     },
//     0001: {
//         damage: 8,
//         name: 'Battle Axe',
//         price: 700, 
//         range: 1,
//         size: 2,
//         weight: 7
//     },
//     0002: {
//         damage: 10,
//         name: 'Great Axe',
//         price: 1400, 
//         range: 1,
//         size: 3,
//         weight: 10
//     },
//     0003: {
//         damage: 10,
//         name: 'Shortbow',
//         price: 2500, 
//         range: [70, 140, 210],
//         size: 2,
//         weight: 2
//     }
// }
    
export default [
    {
        id: '0000',
        damage: 6,
        name: 'Hand Axe',
        price: 400, //copper pieces
        range: 1,
        size: 1, // 1 = Small, 2 = Medium, 3 = Large
        weight: 5 //pounds
    },
    {
        id: '0001',
        damage: 8,
        name: 'Battle Axe',
        price: 700, 
        range: 1,
        size: 2,
        weight: 7
    },
    {
        id: '0002',
        damage: 10,
        name: 'Great Axe',
        price: 1400, 
        range: 1,
        size: 3,
        weight: 10
    },
    {
        id: '0003',
        damage: 10,
        name: 'Shortbow',
        price: 2500, 
        range: [70, 140, 210],
        size: 2,
        weight: 2
    }
]

// const sampleWeapon = {
//     id: 0000,
//     price: 1000, //copper pieces
//     weight: 5, //pounds
//     name: 'hand axe',
//     size: 1, // 1 = Small, 2 = Medium, 3 = Large
//     damage: 6
// }