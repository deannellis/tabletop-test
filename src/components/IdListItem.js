import React from 'react';
import weapons from '../objects/weapons';
import spells from '../objects/spells';

export default (props) => {
    
    const objType = props.obj;
    
    // if(this.props.obj == "weapons") {
    //     objType = 
    // }
    
    const item = weapons.filter((weapon) => {
        return weapon.id === props.id
    });
    
    const spell = spells.filter((spell) => {
        return spell.id === props.id
    });
    
    return (
        <div>
        {objType == 'weapons' && 
            <div>
                <h2>{item[0].name}</h2>
                <p>damage: {item[0].damage}</p>
                <p>price: {item[0].price/100}gp</p>
                <p>weight: {item[0].weight}</p>
                <p>{item[0].range.length && 'ranged'}</p>
            </div>
        }
        {objType == 'spells' && 
            <div>
                <p>{spell[0].name}</p>
            </div>
        }
        </div>
    );
};