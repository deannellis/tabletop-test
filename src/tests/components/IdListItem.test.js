/* global jest expect */
import React from 'react';
import { shallow } from 'enzyme';
import IdListItem from '../../components/IdListItem';
import weapons from '../../objects/weapons';
import spells from '../../objects/spells';

test('should render id list item with spell props', () => {
    const wrapper = shallow(<IdListItem id="0" obj="spells" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render id list item with weapon props', () => {
    const wrapper = shallow(<IdListItem id="0" obj="weapons" />);
    expect(wrapper).toMatchSnapshot();
});