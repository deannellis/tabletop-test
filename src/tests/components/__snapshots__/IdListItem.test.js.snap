// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render id list item with spell props 1`] = `
<div>
  <div>
    <p>
      Charm Person
    </p>
  </div>
</div>
`;

exports[`should render id list item with weapon props 1`] = `
<div>
  <div>
    <h2>
      Hand Axe
    </h2>
    <p>
      damage: 
      6
    </p>
    <p>
      price: 
      4
      gp
    </p>
    <p>
      weight: 
      5
    </p>
    <p />
  </div>
</div>
`;
